mesonbuild/backend/ninjabackend.py

```diff
@@ -2300,12 +2309,17 @@ def generate_pch(self, target, header_deps=None):
             self.add_build(elem)
         return pch_objects
 
+    def get_target_shsym_filename(self, target):
+        # Always name the .symbols file after the primary build output because it always exists
+        targetdir = self.get_target_private_dir(target)
+        return os.path.join(targetdir, target.get_filename() + '.symbols')
+
     def generate_shsym(self, target):
-        target_name = target.get_filename()
         target_file = self.get_target_filename(target)
-        targetdir = self.get_target_private_dir(target)
-        symname = os.path.join(targetdir, target_name + '.symbols')
+        symname = self.get_target_shsym_filename(target)
         elem = NinjaBuildElement(self.all_outputs, symname, 'SHSYM', target_file)
+        # The library we will actually link to, which is an import library on Windows (not the DLL)
```

Comment:
To be pedantic, shouldn't this say that the import library is used with VS (and maybe clang?) rather than Windows in general

Reply:
The ability to link to a DLL is a hack in GCC where it tries to guess what the import library would look like for the DLL (including guessing the calling convention), and so on. It's really bad to do that.

You should _always_ use the import library even with GCC because the import library can contain, f.ex., stubs that refer to multiple DLLs, or constants that aren't present in the DLL at all, etc.

So to be ultra-pedantic, DLLs should never be used for linking!

Reply:
Also, in meson we always use the import library for linking on Windows. We don't have a way to link directly to the DLL at all.